### PR TITLE
Fix EIP712 encodeType with Array

### PR DESCRIPTION
**Pull request by OptimisticQuan**
### What does this PR do?
if types define as below
``` JSON
types: {
  "Group": [
    {
      "name": "name",
      "type": "string"
    },
    {
      "name": "members",
      "type": "Person[]"
    }
  ],
  "Mail": [
    {
      "name": "from",
      "type": "Person"
    },
    {
      "name": "to",
      "type": "Group[]"
    },
    {
      "name": "contents",
      "type": "string"
    }
  ],
  "Person": [
    {
      "name": "name",
      "type": "string"
    },
    {
      "name": "wallets",
      "type": "address[]"
    }
  ]
}
```
`encodeType("Group")` result
expect: `"Group(string name,Person[] members)Person(string name,address[] wallets)"`
actual: `"Group(string name,Person[] members)"`

because `StructuredDataEncoder.getDependencies` does not handle the case where the fields are arrays

This PR contains Fix and Test code.

### Where should the reviewer start?
Changed Files

### Why is it needed?
EIP-712 with Array leads to the wrong result.
This PR fix it.

### CODE REVIEW

File: crypto/src/main/java/org/web3j/crypto/StructuredDataEncoder.java [hunk 1]
```diff
@@ -88,14 +88,15 @@ public Set<String> getDependencies(String primaryType) {
             remainingTypes.remove(remainingTypes.size() - 1);
             deps.add(structName);
 
-            for (StructuredData.Entry entry : types.get(primaryType)) {
-                if (!types.containsKey(entry.getType())) {
+            for (StructuredData.Entry entry : types.get(structName)) {
+                String baseDeclarationTypeName = getBaseTypeNameFromDeclaration(entry.getType());
```
Comment (gtebrean) [inline crypto/src/main/java/org/web3j/crypto/StructuredDataEncoder.java:92]:
```suggestion
                String baseDeclarationTypeName = arrayTypePattern.matcher(declaration).find() ? declaration.substring(0, declaration.indexOf('[')) : declaration;
```
no additional method need it

**Review by gtebrean — COMMENTED** [review]
thanks for investigation, I know was not easy to debug this. Just one comment and if it builds successfully you are good to go.

File: crypto/src/main/java/org/web3j/crypto/StructuredDataEncoder.java (hunk 1 above)
Comment (OptimisticQuan) [reply crypto/src/main/java/org/web3j/crypto/StructuredDataEncoder.java]: ok, I have apply and fix and test the suggested code

Comment (gtebrean) [reply crypto/src/main/java/org/web3j/crypto/StructuredDataEncoder.java]: please run :spotlessJavaApply in order to solve code  style issue

**Review by gtebrean — APPROVED** [review]

**Pull request merged**